faros-ng/basedoc-bundle/3.15/src/Entity/Basedoc/Folder.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Entity\Basedoc;
+
+use App\Repository\Basedoc\FolderRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Faros\Bundle\BasedocBundle\Entity\Folder as FarosFolder;
+use Faros\Bundle\BasedocBundle\Entity\FolderInterface;
+use Gedmo\Blameable\Traits\BlameableEntity;
+use Gedmo\Timestampable\Traits\TimestampableEntity;
+use Symfony\Component\Serializer\Attribute\Groups;
+use Symfony\Component\Serializer\Attribute\MaxDepth;
+
+#[ORM\Entity(repositoryClass: FolderRepository::class)]
+#[ORM\Table(name: 'faros_basedoc_folder')]
+class Folder extends FarosFolder
+{
+    use BlameableEntity;
+    use TimestampableEntity;
+
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column(type: 'integer')]
+    protected ?int $id = null;
+
+    /**
+     * @var Collection<array-key, Document>
+     */
+    #[ORM\ManyToMany(targetEntity: Document::class, mappedBy: 'folders')]
+    #[Groups('basedoc.folder.get')]
+    #[MaxDepth(1)]
+    protected Collection $documents;
+
+    /**
+     * @var Collection<array-key, Folder>
+     */
+    #[ORM\OneToMany(targetEntity: Folder::class, mappedBy: 'parentFolder', cascade: ['remove'])]
+    #[Groups('basedoc.folder.get')]
+    #[MaxDepth(1)]
+    protected Collection $childrenFolders;
+
+    /** @var Folder|null */
+    #[ORM\ManyToOne(targetEntity: Folder::class, cascade: ['persist'], inversedBy: 'childrenFolders')]
+    #[Groups('basedoc.folder.get')]
+    protected ?FolderInterface $parentFolder;
+
+    /**
+     * @var Collection<array-key, FolderAccessRight>
```

Comment:
On peut spécifier le type de la clé maintenant que le type de la clé primaire est aussi défini ici.
```suggestion
     * @var Collection<int, FolderAccessRight>
```